src/Microsoft.AspNetCore.Razor.Language/Intermediate/RazorDiagnosticCollection.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Razor.Language.Intermediate
+{
+    public abstract class RazorDiagnosticCollection : IList<RazorDiagnostic>
```

Comment:
Hmm, thinking about this a little more. Having duplicate RazorDiagnostics doesn't make that much sense. How do you guys feel about this being `ISet<RazorDiagnostic>` instead of `IList<RazorDiagnostic>`?